### PR TITLE
niad-2028-kdev-env-schedule-start-and-stop-for-ecs

### PR DESCRIPTION
**Pull request by Tyler-34**

### CODE REVIEW

File: terraform/aws/modules/module_ecs_service/appautoscaling_schedule.tf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+resource "aws_appautoscaling_scheduled_action" "ecs_schedule_start" {
+  count              = var.enable_ecs_schedule ? 1 : 0  
+  name               = "${local.resource_prefix}_schedule_start"
+  service_namespace  = "ecs"
+  resource_id        = "service/${aws_ecs_service.ecs_service.cluster}/${aws_ecs_service.ecs_service.name}"
+  scalable_dimension = "ecs:service:DesiredCount"
+  schedule           = var.ecs_scheduler_start_time
+
+  scalable_target_action {
+    min_capacity = 1
```
Comment (RIashfaqahmed) [inline terraform/aws/modules/module_ecs_service/appautoscaling_schedule.tf]: min/max values should be taken from env tfvars file. Dont hardcode those with any number

File: terraform/aws/modules/module_ecs_service/appautoscaling_schedule.tf [hunk 2]
```diff
@@ -0,0 +1,27 @@
+resource "aws_appautoscaling_scheduled_action" "ecs_schedule_stop" {
+  count              = var.enable_ecs_schedule ? 1 : 0
+  name               = "${local.resource_prefix}_schedule_stop"
+  service_namespace  = "ecs"
+  resource_id        = "service/${aws_ecs_service.ecs_service.cluster}/${aws_ecs_service.ecs_service.name}"
+  scalable_dimension = "ecs:service:DesiredCount"
```
Comment (RIashfaqahmed) [inline terraform/aws/modules/module_ecs_service/appautoscaling_schedule.tf:6]: is there any setting for desiredcount to 0?

File: terraform/aws/etc/eu-west-2_kdev.tfvars [hunk 3]
```diff
@@ -32,3 +32,6 @@ enable_pss_testbox_scheduler = true
 tag_testbox_enable_scheduler = true
 pss_testbox_scheduler_stop_pattern = "cron(0 00 18 ? * * *)"
 pss_testbox_scheduler_start_pattern = "cron(0 00 06 ? * * *)"
+enable_ecs_scheduler = true
+ecs_scheduler_stop_pattern = "cron(0 00 18 * * ? *)"
+ecs_scheduler_start_pattern = "cron(0 00 06 * * ? *)"
```
Comment (RIashfaqahmed) [inline terraform/aws/etc/eu-west-2_kdev.tfvars]: start pattern should be Mon-Fri daily from 6am-6pm
stop pattern is fine

File: terraform/aws/modules/module_ecs_service/appautoscaling_schedule.tf (hunk 2 above)
Comment (Tyler-34) [reply terraform/aws/modules/module_ecs_service/appautoscaling_schedule.tf:6]: There is no argument for that resource to set desire count.  Changing the min and max capacity affects the desire count to what ever they are set too.

**Review by RIashfaqahmed — DISMISSED** [review]

**Review by ElliotB-96 — APPROVED** [review]

**Pull request merged**